packages/integrations/markdoc/src/heading-ids.ts

```diff
@@ -11,23 +11,17 @@ function getSlug(
 	attributes: Record<string, any>,
 	children: RenderableTreeNode[],
 	headingSlugger: Slugger,
-	experimentalHeadingIdCompat: boolean,
 ): string {
 	if (attributes.id && typeof attributes.id === 'string') {
 		return attributes.id;
 	}
 	const textContent = attributes.content ?? getTextContent(children);
-	let slug = headingSlugger.slug(textContent);
-
-	if (!experimentalHeadingIdCompat) {
-		if (slug.endsWith('-')) slug = slug.slice(0, -1);
-	}
-	return slug;
+	return headingSlugger.slug(textContent);
```

Comment:
Noting that I’m not familiar enough with the Markdoc integration to know if there’s a good way for users to modify our logic if they want backwards compatibility.

We have [docs on using a custom heading component](https://docs.astro.build/en/guides/integrations-guide/markdoc/#custom-headings), but not on how to generate your own `id`.

Maybe they have to write their own `transform()` for headings? We’d need to test.

If it’s super onerous, does anyone think we should offer a backwards compat option for the Markdoc integration? Or do we just embrace that it’s a breaking change and people need to update their links. Markdoc usage is fairly low and the cases where this is breaking will be relatively infrequent.

Reply:
Yeah given how low it is, I don't think it's worth investing a lof of effort in it